pkg/hintrunner/hint.go

```diff
@@ -88,3 +90,78 @@ func (hint TestLessThan) Execute(vm *VM.VirtualMachine) error {
 
 	return nil
 }
+
+type WideMul128 struct {
+	lhs  ResOperander
+	rhs  ResOperander
+	high CellRefer
+	low  CellRefer
+}
+
+func (hint WideMul128) String() string {
+	return "WideMul128"
+}
+
+func (hint WideMul128) Execute(vm *VM.VirtualMachine) error {
+	mask := MaxU128Felt()
+
+	lhs, err := hint.lhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve lhs operand %s: %v", hint.lhs, err)
+	}
+
+	rhs, err := hint.rhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve rhs operand %s: %v", hint.rhs, err)
+	}
+
+	lhsFelt, err := lhs.FieldElement()
+	if err != nil {
+		return err
+	}
+	if lhsFelt.Cmp(&mask) > 0 {
+		return fmt.Errorf("lhs operand %s should be u128", lhsFelt)
+	}
+
+	rhsFelt, err := rhs.FieldElement()
+	if err != nil {
+		return err
+	}
+	if rhsFelt.Cmp(&mask) > 0 {
+		return fmt.Errorf("rhs operand %s should be u128", rhsFelt)
+	}
```

Comment:
Let's do this comparison with `uint256`s. uint256.Cmp is cheaper than fp.Element.Cmp as far as I can see.